src/react/hooks.js

```diff
@@ -48,3 +52,298 @@ export const useTrackColorSchemeChoice = () => {
+          link.remove();
+        });
+      };
+      // find existing links
+      const existingLinks = getExistingCoreThemeLinks();
+
+      // create new link
+      const createCoreThemeLink = (url) => {
+        coreThemeLink = document.createElement('link');
+        coreThemeLink.href = url;
+        coreThemeLink.rel = 'stylesheet';
+        coreThemeLink.dataset.paragonThemeCore = true;
+        coreThemeLink.onload = () => {
+          onLoad();
+          removeExistingLinks(existingLinks);
+        };
+        coreThemeLink.onerror = () => {
+          logError(`Failed to load core theme CSS from ${url}`);
+          if (PARAGON?.themeUrls?.core) {
+            const coreThemeFallbackUrl = `${getConfig().BASE_URL}/${PARAGON.themeUrls.core}`;
+            logInfo(`Falling back to locally installed core theme CSS: ${coreThemeFallbackUrl}`);
+            coreThemeLink = createCoreThemeLink(coreThemeFallbackUrl);
+            const otherExistingLinks = getExistingCoreThemeLinks();
+            removeExistingLinks(otherExistingLinks);
+            document.head.insertAdjacentElement(
+              'afterbegin',
+              coreThemeLink,
+            );
+          }
+        };
+        return coreThemeLink;
+      };
+      coreThemeLink = createCoreThemeLink(themeCore.url);
+      document.head.insertAdjacentElement(
+        'afterbegin',
+        coreThemeLink,
+      );
+    }
+  }, [themeCore?.url, onLoad]);
+};
+
+/**
+ * Adds/updates a `<link>` element in the HTML document to load each theme variant's CSS, setting the
+ * non-current theme variants as "alternate" stylesheets. That is, the browser will still download
+ * the CSS for the non-current theme variants, but at a lower priority than the current theme
+ * variant's CSS. This ensures that if the theme variant is changed at runtime, the CSS for the new
+ * theme variant will already be loaded.
+ *
+ * Note: only "light" theme variant is currently supported.
+ *
+ * @memberof module:React
+ * @param {object} args
+ * @param {object} args.themeVariantUrls An object representing the URLs for each supported theme variant, e.g.: `{ light: 'https://path/to/light.css' }`.
+ * @param {string} args.onLoad A callback function called when the core theme CSS is loaded.
+ */
+const useParagonThemeVariants = ({
+  themeVariants,
+  currentThemeVariant,
+  onLoadThemeVariantLight,
+}) => {
+  useEffect(() => {
+    if (!themeVariants) {
+      return;
+    }
+
+    /**
+     * Determines the value for the `rel` attribute for a given theme variant based
+     * on if its the currently applied variant.
+     */
+    const generateStylesheetRelAttr = (themeVariant) => (currentThemeVariant === themeVariant ? 'stylesheet' : 'alternate stylesheet');
+
+    /**
+     * A helper function to determine which theme variant callback should be used
+     * based on the current theme variant.
+     */
+    const setThemeVariantLoaded = (themeVariant) => {
+      if (themeVariant === 'light') {
```

Comment:
Is there a way we could use the set of theme variant keys in `PARAGON_THEME_URLS` in order to avoid hard-coding 'light', or needing to update frontend-platform when other variants are created?  If a variant doesn't exist, maybe there's just a fallback? 🤔  Seems like we should have enough info on what variants exist to avoid baking them in...

Reply:
See https://github.com/openedx/frontend-platform/pull/440#discussion_r1211073049